### ISSUE TITLE
Adding an error to an already invalid record throws unhandled `becameInvalid` event

### ISSUE THREAD

**Issue by offirgolan** [emberjs/data#3707]
Ember Data > 1.13.6
JSBin: http://emberjs.jsbin.com/tapeva/edit?html,js,output

1. Change ember-data to 1.13.7 or above
2. Type in text field
3. Focus out
4. Type in text field again
5. Focus out
6. Error will be shown in console

`Error: Attempted to handle event 'becameInvalid' on <App.Book:ember418:null> while in state root.loaded.created.invalid.`

[Original StackOverflow Post] (https://stackoverflow.com/questions/32070450/error-attempted-to-handle-event-becameinvalid-while-in-state-root-loaded-crea)

**Comment by thomasdurin** [comment]
Same problem for me. It seems that removing the model's error do not send the event `becameValid`.

**Comment by bmac** [comment]
@tchak I'm not sure what the expected behavior is here. Do you mind taking a look at this issue?

**Comment by offirgolan** [comment]
@bmac i believe the error was introduced through PR #3555 but don't quote me on that.

**Comment by tchak** [comment]
will investigate tonight or tomorrow

**Comment by stefanpenner** [comment]
I tried to replicate in test form, but was unable to. @offirgolan maybe you can use my PR as a stepping stone?

**Comment by offirgolan** [comment]
Okay so after a long time of debugging I figured out what's going on. @stefanpenner couldn't replicate the issue since it was actually fixed but I dont know when or how. Here is my findings:

This is how the flow should be:

1. start in `root.loaded.created.uncommitted` state
2. Add an error which will put the record in `root.loaded.created.invalid` state
3. Change value of `model.firstName` which will trigger `didSetProperty` from the invalid state. This will call `errors.remove` which will put you back in `root.loaded.created.uncommitted` state

This is what was actually happening

1. start in `root.loaded.created.uncommitted` state
2. Add an error which will put the record in `root.loaded.created.invalid` state
3.  Change value of `model.firstName` which will trigger `didSetProperty` from the invalid state. This will call `errors.remove`

In `errors.remove` you are replacing the content of the ArrayProxy (`set(this, 'content', content);`). What is weird about this, is that even though `content.length` is 0,  `this.get('length')` is 1. Because of this `becameValid` is never triggered and the record never gets put back into `root.loaded.created.uncommitted`. It continues to to be in the `root.loaded.created.invalid` state. When you go to try and add another error we again trigger a `becameInvalid` event which doesnt exist under the invalid state.

I was able to replicate the issue up until 2.1.0-beta.1 but once I switched over to 2.2.0-canary, this issue was resolved.
